opentelemetry-android-instrumentation/build.gradle.kts

```diff
@@ -0,0 +1,86 @@
+plugins {
+    id("com.android.library")
+    id("splunk.android-library-conventions")
```

Comment:
This is going to be published as `com.splunk:opentelemetry-android-instrumentation` version 1.x -- while the `internal` package is a really good idea, should we perhaps do anything more to avoid user confusion? E.g. adding `-internal` to the project name, or using `-alpha` version

Reply:
Can we just prevent it from publishing at all? We don't need it to, right?

Reply:
We have to, unfortunately -- since the main `splunk-otel-android` depends on that. (Unless we shade this module into `splunk-otel-android`)

Reply:
Oh yeah, I had expected to shade and probably didn't include that huh. I should work on that.

Reply:
> Oh yeah, I had expected to shade and probably didn't include that huh. I should work on that.

👍 
Do you wanna do that in a separate PR perhaps? I'm guessing it'll involve some gradle magic, and this PR is already large enough.